app/controllers/your-psp/your-psp-tasks.service.test.js

```diff
@@ -0,0 +1,153 @@
+      organisationDetails: true,
+      governmentEntityDocument: false
+    }
+    const activeCredential = getCurrentCredential(account)
+    const taskList = getStripeTaskList(activeCredential, account)
+
+    expect(taskList.ENTER_BANK_DETAILS.enabled).to.equal(true)
+    expect(taskList.ENTER_RESPONSIBLE_PERSON.enabled).to.equal(false)
+    expect(taskList.ENTER_DIRECTOR.enabled).to.equal(false)
+    expect(taskList.ENTER_VAT_NUMBER.enabled).to.equal(false)
+    expect(taskList.ENTER_COMPANY_NUMBER.enabled).to.equal(false)
+    expect(taskList.CONFIRM_ORGANISATION_DETAILS.enabled).to.equal(false)
+    expect(taskList.UPLOAD_GOVERNMENT_ENTITY_DOCUMENT.enabled).to.equal(false)
+
+    expect(taskList.ENTER_BANK_DETAILS.completed).to.equal(false)
+    expect(taskList.ENTER_RESPONSIBLE_PERSON.completed).to.equal(true)
+    expect(taskList.ENTER_DIRECTOR.completed).to.equal(true)
+    expect(taskList.ENTER_VAT_NUMBER.completed).to.equal(true)
+    expect(taskList.ENTER_COMPANY_NUMBER.completed).to.equal(true)
+    expect(taskList.CONFIRM_ORGANISATION_DETAILS.completed).to.equal(true)
+    expect(taskList.UPLOAD_GOVERNMENT_ENTITY_DOCUMENT.completed).to.equal(false)
+
+    expect(stripeTaskListIsComplete(taskList)).to.equal(false)
+  })
+
+  it('should correctly calculate progress for upload govenment document to be enabled only when all the other conditions are met', () => {
+    const account = gatewayAccountFixtures.validGatewayAccount({
+      gateway_account_credentials: [
+        { state: 'ACTIVE', payment_provider: 'stripe', id: 100 }
+      ]
+    })
+    account.connectorGatewayAccountStripeProgress = {
+      bankAccount: true,
+      director: true,
+      vatNumber: true,
+      companyNumber: true,
+      responsiblePerson: true,
+      organisationDetails: true,
+      governmentEntityDocument: false
+    }
+    const activeCredential = getCurrentCredential(account)
+    const taskList = getStripeTaskList(activeCredential, account)
+
+    expect(taskList.ENTER_BANK_DETAILS.enabled).to.equal(false)
+    expect(taskList.ENTER_RESPONSIBLE_PERSON.enabled).to.equal(false)
+    expect(taskList.ENTER_DIRECTOR.enabled).to.equal(false)
+    expect(taskList.ENTER_VAT_NUMBER.enabled).to.equal(false)
+    expect(taskList.ENTER_COMPANY_NUMBER.enabled).to.equal(false)
+    expect(taskList.CONFIRM_ORGANISATION_DETAILS.enabled).to.equal(false)
+
+    expect(taskList.ENTER_BANK_DETAILS.completed).to.equal(true)
+    expect(taskList.ENTER_RESPONSIBLE_PERSON.completed).to.equal(true)
+    expect(taskList.ENTER_DIRECTOR.completed).to.equal(true)
+    expect(taskList.ENTER_VAT_NUMBER.completed).to.equal(true)
+    expect(taskList.ENTER_COMPANY_NUMBER.completed).to.equal(true)
+    expect(taskList.CONFIRM_ORGANISATION_DETAILS.completed).to.equal(true)
+    expect(taskList.UPLOAD_GOVERNMENT_ENTITY_DOCUMENT.completed).to.equal(false)
+
+    expect(stripeTaskListIsComplete(taskList.UPLOAD_GOVERNMENT_ENTITY_DOCUMENT.enabled)).to.equal(true)
```

Comment:
We are not testing method `stripeTaskListIsComplete` here. so assertion should be 

```suggestion
    expect(taskList.UPLOAD_GOVERNMENT_ENTITY_DOCUMENT.enabled).to.equal(true)
```